comp/core/workloadmeta/collectors/internal/ecsfargate/v2parser.go

```diff
@@ -0,0 +1,173 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build docker
+
+// Package ecsfargate implements the ECS Fargate Workloadmeta collector.
+package ecsfargate
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	v2 "github.com/DataDog/datadog-agent/pkg/util/ecs/metadata/v2"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func (c *collector) parseTaskFromV2Endpoint(ctx context.Context) ([]workloadmeta.CollectorEvent, error) {
+	task, err := c.metaV2.GetTask(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return c.parseV2Task(task), nil
+}
+
+func (c *collector) parseV2Task(task *v2.Task) []workloadmeta.CollectorEvent {
+	events := []workloadmeta.CollectorEvent{}
+	seen := make(map[workloadmeta.EntityID]struct{})
+
+	// We only want to collect tasks without a STOPPED status.
+	if task.KnownStatus == workloadmeta.ECSTaskKnownStatusStopped {
+		return events
+	}
+
+	arnParts := strings.Split(task.TaskARN, "/")
+	taskID := arnParts[len(arnParts)-1]
+	entityID := workloadmeta.EntityID{
+		Kind: workloadmeta.KindECSTask,
+		ID:   task.TaskARN,
+	}
+
+	seen[entityID] = struct{}{}
+
+	taskContainers, containerEvents := c.parseTaskContainers(task, seen)
+	entity := &workloadmeta.ECSTask{
+		EntityID: entityID,
+		EntityMeta: workloadmeta.EntityMeta{
+			Name: taskID,
+		},
+		ClusterName: parseClusterName(task.ClusterName),
+		Region:      parseRegion(task.ClusterName),
```

Comment:
nit

I know this was not introduced in your PR, but mentioning it in case you have time to improve things a bit.

you can already use the AWS public go sdk to get the region instead of manually parsing the ARN.

[Reference](https://github.com/aws/aws-sdk-go/blob/49815a2f43cdd9dd46524ccdc8106615e63cf01c/aws/arn/arn.go#L61-L76)

Reply:
Good point 👍